test/integration/client-side-encryption/client_side_encryption.prose.25.lookup.test.ts

```diff
@@ -0,0 +1,413 @@
+       *   db.csfle with options: { "validator": { "$jsonSchema": "<schema-csfle.json>"}}.
+       *   db.csfle2 with options: { "validator": { "$jsonSchema": "<schema-csfle2.json>"}}.
+       *   db.qe with options: { "encryptedFields": "<schema-qe.json>"}.
+       *   db.qe2 with options: { "encryptedFields": "<schema-qe2.json>"}.
+       *   db.no_schema with no options.
+       *   db.no_schema2 with no options.
+       * ```
+       */
+      const collections = [
+        {
+          name: 'csfle',
+          options: { validator: { $jsonSchema: await readFixture('schema-csfle.json') } },
+          document: { csfle: 'csfle' }
+        },
+        {
+          name: 'csfle2',
+          options: { validator: { $jsonSchema: await readFixture('schema-csfle2.json') } },
+          document: { csfle2: 'csfle2' }
+        },
+        {
+          name: 'qe',
+          options: { encryptedFields: await readFixture('schema-qe.json') },
+          document: { qe: 'qe' }
+        },
+        {
+          name: 'qe2',
+          options: { encryptedFields: await readFixture('schema-qe2.json') },
+          document: { qe2: 'qe2' }
+        },
+        {
+          name: 'no_schema',
+          options: {},
+          document: { no_schema: 'no_schema' }
+        },
+        {
+          name: 'no_schema2',
+          options: {},
+          document: { no_schema2: 'no_schema2' }
+        }
+      ];
+
+      for (const { name, options } of collections) {
+        await client.db('db').createCollection(name, options);
+      }
+
+      /**
+       * Create an encrypted MongoClient configured with:
+       *
+       * ```txt
+       *   AutoEncryptionOpts(
+       *       keyVaultNamespace="db.keyvault",
+       *       kmsProviders={"local": { "key": "<base64 decoding of LOCAL_MASTERKEY>" }}
+       *   )
+       * ```
+       */
+      encryptedClient = newEncryptedClient(this);
+
+      /**
+       * ```
+       * {"csfle": "csfle"} into db.csfle
+       * Use the unencrypted client to retrieve it. Assert the csfle field is BSON binary.
+       * {"csfle2": "csfle2"} into db.csfle2
+       * Use the unencrypted client to retrieve it. Assert the csfle2 field is BSON binary.
+       * {"qe": "qe"} into db.qe
+       * Use the unencrypted client to retrieve it. Assert the qe field is BSON binary.
+       * {"qe2": "qe2"} into db.qe2
+       * Use the unencrypted client to retrieve it. Assert the qe2 field is BSON binary.
+       * {"no_schema": "no_schema"} into db.no_schema
+       * {"no_schema2": "no_schema2"} into db.no_schema2
+       * ```
+       */
+      for (const { name, document } of collections) {
+        const { insertedId } = await encryptedClient.db('db').collection(name).insertOne(document);
+
+        if (name.startsWith('no_')) continue;
+
+        expect(await client.db('db').collection(name).findOne(insertedId))
+          .to.have.property(Object.keys(document)[0])
+          .that.has.property('_bsontype', 'Binary');
```

Comment:
How would you feel about removing the looping in favor of simply creating each collection explicitly and inserting documents explicitly?  I think complexity-wise it's about the same but the explicit approach is easier to compare with the spec test line-by-line.

Reply:
I wouldn't be in favor, loops are a reasonable reduction of verbatim code complexity, and table testing is a common pattern so you know X is applied the same way to all inputs, rather than missing a small deviation between actual code paths.

Reply:
> loops are a reasonable reduction of verbatim code complexity

But here all we should need is 6 create collections calls, and 4 insert calls?  I don't think a loop reduces much complexity here.  And as a reviewer, comparing this with the spec, I'd much rather compare line-by-line than need to cross reference the spec, the table, and the multiple loops used here.

> table testing is a common pattern so you know X is applied the same way to all inputs
Table testing is great for pure input -> output testing but breaks down in its efficacy when the test logic isn't identical for each test case.  In this case (and the table tests below), the loop has different logic for different tests, which obfuscates both the intent of the table testing matrix and the actual test logic.